packages/components/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { forwardRef, useEffect, useRef } from "react";
+import { Animated, Easing, View, ViewProps } from "react-native";
+import { useStyles } from "../../hooks/useStyles";
+import { useToken } from "../../hooks/useToken";
+import { EDSStyleSheet } from "../../styling";
+import { Paper } from "../Paper";
+import { PressableHighlight } from "../PressableHighlight";
+export type SwitchProps = {
+    onChange?: (isActive: boolean) => void;
+    color?: "primary" | "secondary" | "danger";
+    active?: boolean;
+    disabled?: boolean;
+    loading?: boolean;
+};
+
+const KNOB_SIZE = 20;
+const WIDTH = 60;
+const HEIGHT = 60;
+
+export const Switch = forwardRef<View, SwitchProps & ViewProps>(
+    (
+        {
+            color = "primary",
+            onChange = () => null,
+            active = false,
+            disabled = false,
+            loading = false,
+            ...rest
+        },
+        ref,
+    ) => {
+        const styles = useStyles(themeStyles, {
+            color,
+            disabled,
+            isActive: active,
+        });
+
+        const token = useToken();
+
+        const knobProgressValue = useRef(
+            new Animated.Value(active ? WIDTH - KNOB_SIZE * 1.7 : 0),
+        ).current;
+
+        const backgroundProgressValue = useRef(new Animated.Value(0)).current;
+
+        const activeKnobAnimation = Animated.timing(knobProgressValue, {
+            toValue: 1,
+            duration: token.timing.animation.slow,
+            useNativeDriver: true,
+            easing: Easing.inOut(Easing.ease),
+        });
+
+        const inactiveKnobAnimation = Animated.timing(knobProgressValue, {
+            toValue: 0,
+            duration: token.timing.animation.slow,
+            useNativeDriver: true,
+            easing: Easing.inOut(Easing.ease),
+        });
+
+        const displacement = knobProgressValue.interpolate({
+            inputRange: [0, 1],
+            outputRange: [0, WIDTH - KNOB_SIZE * 1.7],
+            extrapolate: "clamp",
+        });
+
+        const activeBackgroundAnimation = Animated.timing(backgroundProgressValue, {
+            toValue: 1,
+            duration: token.timing.animation.slow,
+            useNativeDriver: true,
+            easing: Easing.inOut(Easing.ease),
+        });
+
+        const inactiveBackgroundAnimation = Animated.timing(backgroundProgressValue, {
+            toValue: 0,
+            duration: token.timing.animation.slow,
+            useNativeDriver: true,
+            easing: Easing.inOut(Easing.ease),
+        });
+
+        const backgroundDisplacement = backgroundProgressValue.interpolate({
+            inputRange: [0, 1],
+            outputRange: [0.001, 1],
```

Comment:
Why was this value set to 0.001 ?

Reply:
it was based on some old version where the value had to be greater than zero, it will be changed